contracts/factories/ManagerFactory.sol

```diff
@@ -0,0 +1,201 @@
+
+import { DelegatedManager } from "../manager/DelegatedManager.sol";
+import { IDelegatedManager } from "../interfaces/IDelegatedManager.sol";
+import { ISetTokenCreator } from "../interfaces/ISetTokenCreator.sol";
+
+contract ManagerFactory {
+    struct InitializeParams{
+        address deployer;
+        address owner;
+        IDelegatedManager manager;
+        bool isPending;
+    }
+    
+    ISetTokenCreator public factory;
+    mapping(ISetToken=>InitializeParams) public initializeState;
+    mapping(ISetToken=>bool) public isValidSet;
+    address[] internal validSets; 
+
+    constructor(
+        ISetTokenCreator _setTokenFactory
+    ) public {
+        factory = _setTokenFactory;
+    }
+
+    /* ============ External Functions ============ */
+
+    function createSetAndManager(
+        address[] memory _components,
+        int256[] memory _units,
+        string memory _name,
+        string memory _symbol,
+        address _owner,
+        address _methodologist,
+        address[] memory _modules,
+        address[] memory _operators,
+        address[] memory _assets,
+        address[] memory _extensions
+    )
+        external
+        returns (ISetToken, address)
+    {
+        require(_extensions.length > 0, "Must have at least 1 extension");
+        // Require that components of Set are contained in the _assets array
+        // Require there be at least one operator?
+
+        ISetToken setToken = _deploySet(
+            _components,
+            _units,
+            _modules,
+            _name,
+            _symbol
+        );
+
+        DelegatedManager manager = _deployManager(
+            setToken,
+            _methodologist,
+            _extensions,
+            _operators,
+            _assets
+        );
+
+        _setInitializationState(setToken, address(manager), _owner);
+
+        setToken.setManager(address(manager));
+
+        return (setToken, address(manager));
+    }
+
+    function createManager(
+        ISetToken _setToken,
+        address _owner,
+        address _methodologist,
+        address[] memory _operators,
+        address[] memory _assets,
+        address[] memory _extensions
+    )
+        external
+        returns (address)
+    {
+        DelegatedManager manager = _deployManager(
+            _setToken,
+            _methodologist,
+            _extensions,
+            _operators,
+            _assets
+        );
+
+        _setInitializationState(_setToken, address(manager), _owner);
+
+        return address(manager);
+    }
+
+    function initialize(
+        ISetToken _setToken,
+        address[] memory _initializeTargets,
+        bytes[] memory _initializeBytecode
+    )
+        external
+    {
+        require(initializeState[_setToken].isPending, "Manager must be awaiting initialization");
+        require(msg.sender == initializeState[_setToken].deployer, "Only deployer can initialize manager");
```

Comment:
Should we add a check that `_initializeTargets.validatePairsWithArray(_initializeBytecode)` (from the AddressArrays lib)?